test/integration/smart_answer_flows/coronavirus_find_support_test.rb

```diff
@@ -0,0 +1,2540 @@
+require_relative "../../test_helper"
+require_relative "flow_test_helper"
+
+require "smart_answer_flows/coronavirus-find-support.rb"
+
+class CoronavirusFindSupportFlowTest < ActiveSupport::TestCase
+  include FlowTestHelper
+
+  setup do
+    setup_for_testing_flow SmartAnswer::CoronavirusFindSupportFlow
+  end
+
+  context "all combinations of all options" do
```

Comment:
I don't think all the options need to be tested here. Following a discussion with Graham, I think there are 7 key questions that lead to changes in the user journey through the flow. Each of these 7 is independent of the others (if user selects X they are taken through subsection x no matter how they answered questions Y and/or Z). I would suggest therefore that the following flows are tested:

- One each for the 7 options feeling_unsafe paying_bills getting_food being_unemployed going_to_work, somewhere_to_live and mental_health, where only that option is selected and therefore only a single subsection of the flow is inserted between the start and the end of the flow.
- One for the journey where none of the options are selected.
- One for the journey where all the options are selected (to check they combine together corrently)

Reply:
Thanks @reggieb 👍  

Generally, I feel if you can test all possible options, then you should.  But I totally understand that isn't always a pragmatic approach.  

I'm not 100% happy to remove all but the 9 tests you indicate, as I feel we are no longer testing all paths and transitions which could expose us to 'incorrect' passing tests.  I don't want to create the potential for subtle and difficult to fix and find bugs!

Having said that, I do see the issue with the volume of tests and the maintenance of them, and I share your concerns on that.

Can we compromise?  A single test - that covers all 128 paths and transitions, it will also handle all changes we might make to the flow, the `next_node` blocks and the questions.  Of course, this approach requires a loop that goes around each possible option, transition and test them in-turn - this is how I generated the 128 tests in the first instance - you didn't think I typed them all in right ;-)  I know this type of test is more difficult to maintain and debug, and requires sufficient output to identify a failing test when one is generated.  I'm also aware that some may find this approach undesirable - it's probably not the "GDS way"!  However, in this instance it may be the best - most pragmatic approach.

Reply:
OK - really controversial. How about you wrapping up the script you used to generate this spec, in a rake task - so that it can be easily replicated and updated. Would that be an acceptable compromise? That way you get the coverage of all happy journey plus a way to maintain it.

Reply:
More than happy with that approach - good shout!  Will add it 👍

Reply:
Hi @reggieb I've added the test generation script.  

Initially, I attempted to create a generic script to create tests for all combinations of any and all smart answers.  While I feel this is possible and I got a fair way to doing it, I don't feel it should be done as part of this story and will take more time than I feel we have right now.

Reply:
Sorry I'm late to this thread - I few a concerns about having integration tests for every path.
- Integration tests are expensive - how much extra time does it take to run these test?
- Is the added complexity of rake task to generate the test file worth the benefit we have from these extra tests?
- How do future developer know to run the rake task, rather than update the test file directly?
- Does this scale? What happens if we were to add another more questions or options? I can see us easily ending up with 1000+ scenarios if we were to add a more questions or options.
- We don't test every possible case in other smart answer flows, adding inconsistency in how we write tests, is this going to be confusing for future developers?
- What are we testing here exactly? Is it the next_node routing logic? If we made a help function (suggest in a comment above) - we could write unit tests to cover that more comprehensively.